cibuildwheel/macos.py

```diff
@@ -560,9 +562,12 @@ def build(options: Options, tmp_path: Path) -> None:
                     call_with_arch = functools.partial(call, *arch_prefix)
                     shell_with_arch = functools.partial(call, *arch_prefix, "/bin/sh", "-c")
 
-                    # Use --no-download to ensure determinism by using seed libraries
-                    # built into virtualenv
-                    call_with_arch("python", "-m", "virtualenv", "--no-download", venv_dir, env=env)
+                    # Use pip version from the initial env to ensure determinism
+                    venv_args = ["--no-periodic-update", f"--pip={pip_version}"]
+                    # In Python<3.12, setuptools & wheel are installed as well, use virtualenv embedded ones
+                    if Version(config.version) < Version("3.12.0a0"):
```

Comment:
Why 3.12.0a0? 3.12 would be fine, we don’t have pre-releases for 3.12 and this is just a two element version anyway?

Reply:
It's just a habit to exclude pre-release also, but yeah, I can change that.